modules/db/src/main/java/org/mule/module/db/internal/domain/type/ClobResolvedDataType.java

```diff
@@ -29,26 +34,31 @@ public ClobResolvedDataType(int id, String name)
     @Override
     public void setParameterValue(PreparedStatement statement, int index, Object value) throws SQLException
     {
-        if (value != null && !(value instanceof Clob))
-        {
-            if (value instanceof String)
-            {
-                statement.setCharacterStream(index, new StringReader((String) value), ((String) value).length());
-            }
-            else if (value instanceof InputStream)
-            {
-                String stringValue = IOUtils.toString((InputStream) value);
-                statement.setCharacterStream(index, new StringReader(stringValue), stringValue.length());
-            }
-            else
-            {
+        if (value != null && !(value instanceof Clob)) {
+            String valueString;
+
+            if (value instanceof String) {
+                valueString = (String) value;
+            } else if (value instanceof InputStream) {
+                valueString = IOUtils.toString((InputStream) value);
```

Comment:
Clob have a setCharacterStream method. In case of value is instanceOf inputStream you should use this method.